### PR TITLE
Update docs to mention gitignore changes

### PR DESCRIPTION
**Pull request by moiri-gamboni**
## Description

Updates docs to mention that prettier follows .gitignore in the CWD.

## Checklist

- [ ] I’ve added tests to confirm my change works.
- [ ] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory).
- [ ] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/main/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

**Comment by awsr** [comment]
Any chance of getting this merged? I spent hours trying to figure out why Prettier wasn't working only to finally stumble upon the answer on my own because searching the documentation for ignoring code showed _nothing_ about it being used by default. It only gets mentioned elsewhere under the description for the `--ignore-path` path option (which I wasn't using, so why would I look there?), and even _that_ **doesn't show up in the results** when searching for "gitignore" in the documentation.

**Review by sosukesuzuki — APPROVED** [review]

File: docs/ignore.md [hunk 1]
```diff
@@ -24,7 +24,7 @@ coverage
 
 It’s recommended to have a `.prettierignore` in your project! This way you can run `prettier --write .` to make sure that everything is formatted (without mangling files you don’t want, or choking on generated files). And – your editor will know which files _not_ to format!
 
-By default prettier ignores files in version control systems directories (".git", ".svn" and ".hg") and `node_modules` (if [`--with-node-modules` CLI option](cli.md#--with-node-modules) not specified)
+By default prettier ignores files in version control systems directories (".git", ".svn" and ".hg") and `node_modules` (if [`--with-node-modules` CLI option](cli.md#--with-node-modules) not specified). Prettier will also follow rules specified in .gitignore if it exists in the same directory from which it is run.
```
Comment (fisker) [inline docs/ignore.md:27]: We also ignores `.sl`  https://github.com/prettier/prettier/blob/56945ecf12a5b2e3b70e13bbcec956ccaf7bd460/src/cli/expand-patterns.js#L47

Comment (fisker) [reply docs/ignore.md:27]: If we are going to mention `.gitignore` here, maybe we should put `.prettierignore` here too.

Comment (fisker) [inline docs/ignore.md:27]:
```suggestion
By default prettier ignores files in version control systems directories (".git", ".sl", ".svn", and ".hg") and `node_modules` (unless [`--with-node-modules` CLI option](cli.md#--with-node-modules) is specified). Prettier will also follow rules specified in ".gitignore" file if it exists in the same directory from which it is run.
```

Comment (awsr) [inline docs/ignore.md:27]: Slightly better wording would probably be:

```suggestion
By default prettier ignores files in version control systems directories (".git", ".sl", ".svn" and ".hg") and `node_modules` (unless the [`--with-node-modules` CLI option](cli.md#--with-node-modules) is specified). Prettier will also follow rules specified in the ".gitignore" file if it exists in the same directory from which it is run.
```

**Review by fisker — APPROVED** [review]

**Comment by fisker** [comment]
Hello @moiri-gamboni Can you copy change to files in `website/versioned_docs/version-stable/`? So the website can be updated when we merge.

**Pull request merged**